plugins/hls-explicit-imports-plugin/src/Ide/Plugin/ExplicitImports.hs

```diff
@@ -71,33 +87,36 @@ descriptorForModules recorder pred plId =
   (defaultPluginDescriptor plId)
     {
       -- This plugin provides a command handler
-      pluginCommands = [importLensCommand],
+      pluginCommands = [PluginCommand importCommandId "Explicit import command" runImportCommand],
       -- This plugin defines a new rule
-      pluginRules = minimalImportsRule recorder,
-      pluginHandlers = mconcat
-        [ -- This plugin provides code lenses
-          mkPluginHandler SMethod_TextDocumentCodeLens $ lensProvider pred
+      pluginRules = minimalImportsRule recorder pred,
+      pluginHandlers =
+         -- This plugin provides code lenses
+           mkPluginHandler SMethod_TextDocumentCodeLens (lensProvider pred)
+        <> mkPluginHandler SMethod_CodeLensResolve lensResolveProvider
           -- This plugin provides code actions
-        , mkPluginHandler SMethod_TextDocumentCodeAction $ codeActionProvider pred
-        ]
-    }
+        <> mkCodeActionHandlerWithResolve (codeActionProvider pred) codeActionResolveProvider
 
--- | The command descriptor
-importLensCommand :: PluginCommand IdeState
-importLensCommand =
-  PluginCommand importCommandId "Explicit import command" runImportCommand
-
--- | The type of the parameters accepted by our command
-newtype ImportCommandParams = ImportCommandParams WorkspaceEdit
-  deriving (Generic)
-  deriving anyclass (FromJSON, ToJSON)
+    }
 
 -- | The actual command handler
-runImportCommand :: CommandFunction IdeState ImportCommandParams
-runImportCommand _state (ImportCommandParams edit) = do
-  -- This command simply triggers a workspace edit!
-  _ <- sendRequest SMethod_WorkspaceApplyEdit (ApplyWorkspaceEditParams Nothing edit) (\_ -> pure ())
-  return (Right Null)
+runImportCommand :: CommandFunction IdeState EIResolveData
+runImportCommand ideState eird = pluginResponse $ do
+  case eird of
+    ResolveOne uri int -> do
```

Comment:
Missing case for `ResolveAll`?

Reply:
Also, since you've named the fields, this can be a nice place to use `NamedFieldPuns`, i.e.
```
ResolveOne{uri, int} -> ...
```
just makes sure you can't mix up the positional args

Reply:
The command actually is responding only to the lens, so shouldn't handle the ResolveAll case. In this case I'm actually reusing the data function, but it might be better to have seperate types.

Reply:
I think it seems reasonable to share the code for running the action if possible? Having one function that handles either case and then we just call it seems good.

Reply:
Okay good point, there is no reason to use resolve from codeActions then, but no reason to implement resolve for no reason.